metricbeat/module/system/filesystem/filesystem.go

```diff
@@ -66,26 +65,31 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 // Fetch fetches filesystem metrics for all mounted filesystems and returns
 // an event for each mount point.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) {
 	fss, err := GetFileSystemList()
 	if err != nil {
-		return nil, errors.Wrap(err, "filesystem list")
+		r.Error(errors.Wrap(err, "filesystem list"))
+		return
 	}
 
 	if len(m.config.IgnoreTypes) > 0 {
 		fss = Filter(fss, BuildTypeFilter(m.config.IgnoreTypes...))
 	}
 
-	filesSystems := make([]common.MapStr, 0, len(fss))
 	for _, fs := range fss {
 		fsStat, err := GetFileSystemStat(fs)
 		if err != nil {
 			debugf("error getting filesystem stats for '%s': %v", fs.DirName, err)
 			continue
 		}
 		AddFileSystemUsedPercentage(fsStat)
-		filesSystems = append(filesSystems, GetFilesystemEvent(fsStat))
-	}
 
-	return filesSystems, nil
+		event := mb.Event{
+			MetricSetFields: GetFilesystemEvent(fsStat),
+		}
+		if !r.Event(event) {
+			debugf("Failed to report event, interrupting Fetch")
```

Comment:
Why this is a debug msg instead of error?

Reply:
This is actually expected. If `r.Event()` returns false, it means that it couldn't send the event because the metricset is being closed. If it is being closed it is ok to don't continue trying to send events, specially in this case that to generate each event some information is requested to the system.

We had a discussion about the use of the return value of `r.Event()` [here](https://github.com/elastic/beats/pull/9907#discussion_r245977781).